core/shared/src/main/scala-2/zio/internal/macros/LayerMacroUtils.scala

```diff
@@ -84,18 +84,20 @@ private[zio] trait LayerMacroUtils {
   }
 
   private def buildFinalTree(tree: LayerTree[LayerExpr]): LayerExpr = {
-    val memoMap: Map[LayerExpr, LayerExpr] =
+    val memoList: List[(LayerExpr, LayerExpr)] =
       tree.toList.map { node =>
         val freshName = c.freshName("layer")
         val termName  = TermName(freshName)
         node -> c.Expr(q"$termName")
-      }.toMap
+      }
```

Comment:
Should we do a `.distinct` after the `toList` just to make sure we don't have duplicates or no need?

Reply:
```LayerTree#toList``` still does a fold into a ```Set``` (a ```LinkedHashSet``` this time, to keep an order), thus producing a distinct output, so no need for an explicit call to ```.distinct```